### ISSUE TITLE
Rule Change: no-useless-constructor should permit constructor with changed required parameters

### ISSUE THREAD

**Issue by gthb** [eslint/eslint#19291]
### What rule do you want to change?

no-useless-constructor

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
class Foo {
  constructor (arg?: string) {
    console.log(arg);
  }
}

class Bar extends Foo {
  constructor (arg: string) {
    super(arg);
  }
}
```

### What does the rule currently do for this code?

Currently the rule flags the constructor of `Bar` as a “Useless constructor”.

But this constructor prohibits instantiating without the `arg` parameter, changing instantiation behavior vs the superclass. I think that isn't useless.

### What will the rule do after it's changed?

Refrain from flagging a constructor that overrides its superclass constructor with a different parameter list (in this case changing an optional parameter to a required one, but I bet this applies also to other kinds of type narrowing).

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @gthb, thanks for the issue!

> Refrain from flagging a constructor that overrides its superclass constructor with a different parameter list (in this case changing an optional parameter to a required one, but I bet this applies also to other kinds of type narrowing).

Comparing parameter lists would require type information, which isn't available in core rules.

This could be done in the [`@typescript-eslint/no-useless-constructor`](https://typescript-eslint.io/rules/no-useless-constructor/) extension rule, but it seems that a similar request wasn't accepted:

https://github.com/typescript-eslint/typescript-eslint/issues/3820

**Comment by gthb** [comment]
Oh right, makes sense. Thanks for your response.
